server/routes/groupRoutes.js

```diff
@@ -139,5 +139,104 @@ router.post('/:groupName/change/:key', (req, res) => {
   });
 });
 
+
+//delete a group
+//TODO: make sure only admin of group can delete a group
+router.post('/:groupName/delete', (req, res) => {
+  //req.user.username --> check if this is admin
+
+  let groupName = req.params.groupName;
+  db.Group.destroy({
+    where: {
+      groupName: groupName
+    }
+  }).then(([numRows, rowsAffected]) => {
+    if (numRows === 0) {
+      res.status(404).json({ err: 'group not found' });
+      return;
+    }
+  }).catch((err) => {
+    console.log('Error while retrieving group.');
+    console.log(err);
+    res.status(500).json({ err: err });
+    return;
+  });
+});
+
+//add a user to a group
+//TODO: make sure only admin of group can add user to a group
+router.post('/:groupName/add', (req, res) => {
+  let groupName = req.params.groupName;
+  let username = req.body.username;
+  let group = db.Group.findByPk(req.params.groupName).then((modelInstance) => {
```

Comment:
is this async?

Reply:
I agree that if this function is async, `let group = ` should be removed?

Reply:
idk if it is async!